### ISSUE TITLE
forgotten SONAME bump?

### ISSUE THREAD

**Issue by mones** [dinhviethoa/libetpan#249]
Hi!

While trying to package latest release version for Debian just noticed that there's some symbols which have been removed from libetpan since previous releases:
```
+#MISSING: 1.7-1# MD5Final@Base 1.0
+#MISSING: 1.7-1# MD5Init@Base 1.0
+#MISSING: 1.7-1# MD5Update@Base 1.0
+#MISSING: 1.7-1# hmac_md5@Base 1.0
+#MISSING: 1.7-1# hmac_md5_final@Base 1.0
+#MISSING: 1.7-1# hmac_md5_import@Base 1.0
+#MISSING: 1.7-1# hmac_md5_init@Base 1.0
+#MISSING: 1.7-1# hmac_md5_precalc@Base 1.0
+#MISSING: 1.7-1# mailimap_oauth2_authenticate_send@Base 1.4
+#MISSING: 1.7-1# mailimf_fws_atom_for_word_parse@Base 1.6
```
But the SONAME has not changed (the file version has, though):
```
$ objdump -p /usr/lib/x86_64-linux-gnu/libetpan.so.17.1.0 | grep SONAME # previous
  SONAME               libetpan.so.17
$ objdump -p libetpan.so.17.2.0 | grep SONAME # latest
  SONAME               libetpan.so.17
```

I believe SONAME should be bumped when changes which could break existing library clients are shipped in new versions.

**Comment by dinhviethoa** [comment]
https://github.com/dinhviethoa/libetpan/commit/da697653cee565deaa3316c72f9cf47a211d9407
Did I get it wrong?

**Comment by mones** [comment]
Commented on the commit by mistake, I hope you don't mind ;-)

**Comment by dinhviethoa** [comment]
Fixed. Let me know if everything looks good for you.

**Comment by mones** [comment]
Yes, this should fix this issue, but unfortunately not everything looks good :crying_cat_face:

The latest release is tagged 1.7.2 but `make dist` is generating a 1.7 tarball:

`-rw-r--r--  1 devel devel 1306720 Jun  8 12:20 libetpan-1.7.tar.gz`

This, by itself, it's not a real problem, since I can repackage and rename the tarball to 1.7.2. Nevertheless, I've noticed that build system doesn't support micro versions at all, so, for example running `libetpan-config --version` echoes just 1.7 instead 1.7.2, which would be expected now.

If, for example, tomorrow a 1.7.3 is released, it will still say it's 1.7, which doesn't look good for any program trying to rely on libetpan version.

So, the options are clear. The easy one: just increase minor version when API changes (for this one 1.8 instead of 1.7.2) or, a bit more work: add support for micro versions in configure.ac.
